src/v0/destinations/marketo_static_list/util.js

```diff
@@ -34,7 +35,47 @@ const validateMessageType = (message, allowedTypes) => {
   }
 };
 
+function transformForRecordEvent(inputs, leadIdObj) {
+  const finalMetadata = [];
+  // iterate through each inputs metadata and create a final metadata
+  const tokenisedInputs = inputs.map((input) => {
+    const { message } = input;
+    const { metadata } = input;
+    finalMetadata.push(metadata);
+    const { fields, action, type } = message;
+    if (type !== 'record') {
+      throw new InstrumentationError('Invalid message type, Supported message type is record.');
+    }
+    const { properties } = message;
+    if (!properties) {
+      message.properties = {};
+    }
+    message.properties.listData = message?.properties.listData || { add: [], remove: [] };
+    const fieldsId = fields?.id;
+    if (fieldsId === undefined) {
+      throw new InstrumentationError('No lead id passed in the payload.');
+    }
+    if (action === 'insert') {
+      leadIdObj.insert.push({ id: fieldsId });
+      message.properties.listData.add.push({ id: fieldsId });
+    } else if (action === 'delete') {
+      leadIdObj.delete.push({ id: fieldsId });
+      message.properties.listData.remove.push({ id: fieldsId });
+    } else {
+      throw new InstrumentationError('Invalid action type');
+    }
+    return input;
+  });
+  const finalInput = [tokenisedInputs[0]];
```

Comment:
What if inputs is of size 0? Does this crash with index out of bounds error?

Reply:
Handled the check in transform.js at [L141](https://github.com/rudderlabs/rudder-transformer/blob/bb41537b070db1990d1879a7881c227459e0336e/src/v0/destinations/marketo_static_list/transform.js#L141) and [L149](https://github.com/rudderlabs/rudder-transformer/blob/bb41537b070db1990d1879a7881c227459e0336e/src/v0/destinations/marketo_static_list/transform.js#L149)